cpp/src/reductions/scan/rank_scan.cu

```diff
@@ -32,6 +32,22 @@ namespace cudf {
 namespace detail {
 namespace {
 
+template <typename device_comparator_type, typename value_resolver>
+struct rank_equality_functor {
+  rank_equality_functor(device_comparator_type comparator, value_resolver resolver)
+    : _comparator(comparator), _resolver(resolver)
+  {
+  }
+
+  auto __device__ operator()(size_type row_index)
+  {
+    return _resolver(row_index == 0 || !_comparator(row_index, row_index - 1), row_index);
+  }
+
+  device_comparator_type _comparator;
+  value_resolver _resolver;
```

Comment:
By convention, struct members (public access) should not have underscore prefix. You only add prefix for class members, or struct members that are private. I forgot to check this for the code above, so this also applies everywhere in this PR.
```suggestion
  device_comparator_type comparator;
  value_resolver resolver;
```

Reply:
I added these because another reviewer @hyperbolic2346 suggested they should be there according the contribution guidelines.

Reply:
I will remove the underscores and remove the explicit constructor. I'll do these for all functors - making them more readable.

Reply:
Would you feel more comfortable if I made the members `private` to follow the naming convention as it exists now?

Reply:
Functors like this should have public members so you can avoid declaring a constructor. Otherwise, you have to write explicit constructor for it.

Reply:
Since this struct already has a constructor, it's up to you to decide to keep the members private.

Reply:
I pushed a commit making the members private.